File: hw/palomo/u3/hw26Observer/Observer.js/src/controller/StockController.js
/**
 * @author Cristian Palomo, Error 404, @ESPE
 */
import { Financier } from "../model/Financier.js";
import { IBM } from "../model/IBM.js";

export class StockController {
    startApplication() {
        const s = new Financier("Sorros");
        const b = new Financier("Berkshire");

        const ibm = new IBM("IBM", 120.0);
        ibm.addObserver(s);
        ibm.addObserver(b);

        ibm.setPrice(120.1);
        ibm.setPrice(121.0);
        ibm.setPrice(120.5);
        ibm.setPrice(120.75);
        ibm.setSymbol("IBMTEST");
    }
}

new StockController().startApplication();

File: hw/palomo/u3/hw26Observer/Observer/src/main/java/ec/edu/espe/observer/controller/StockController.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.observer.controller;

import ec.edu.espe.observer.model.IBM;
import ec.edu.espe.observer.model.Financier;
/**
 *
 * @author Cristian Palomo, Error 404, @ESPE
 */
public class StockController {
    public void startApplication() {
        Financier s = new Financier("Sorros");
        Financier b = new Financier("Berkshire");

        IBM ibm = new IBM("IBM", 120.00);
        ibm.addObserver(s);
        ibm.addObserver(b);

        ibm.setPrice(120.10);
        ibm.setPrice(121.00);
        ibm.setPrice(120.50);
        ibm.setPrice(120.75);
        ibm.setSymbol("IBMTEST");
    }

    public static void main(String[] args) {
        new StockController().startApplication();
    }
}
